gradio/external.py

```diff
@@ -109,369 +100,271 @@ def load_blocks_from_repo(
+    # Example: distilbert-base-uncased-finetuned-sst-2-english
+    elif p == "text-classification":
+        inputs = components.Textbox(label="Input")
+        outputs = components.Label(label="Classification")
+        examples = ["I feel great"]
+        postprocess = external_utils.postprocess_label
+        fn = client.text_classification
+    # Example: gpt2
+    elif p == "text-generation":
+        inputs = components.Textbox(label="Text")
+        outputs = inputs
+        examples = ["Once upon a time"]
+        fn = external_utils.text_generation_wrapper(client)
+    # Example: valhalla/t5-small-qa-qg-hl
+    elif p == "text2text-generation":
+        inputs = components.Textbox(label="Input")
+        outputs = components.Textbox(label="Generated Text")
+        examples = ["Translate English to Arabic: How are you?"]
+        fn = client.text_generation
+    # Example: Helsinki-NLP/opus-mt-en-ar
+    elif p == "translation":
+        inputs = components.Textbox(label="Input")
+        outputs = components.Textbox(label="Translation")
+        examples = ["Hello, how are you?"]
+        fn = client.translation
+    # Example: facebook/bart-large-mnli
+    elif p == "zero-shot-classification":
+        inputs = [
+            components.Textbox(label="Input"),
+            components.Textbox(label="Possible class names (" "comma-separated)"),
+            components.Checkbox(label="Allow multiple true classes"),
+        ]
+        outputs = components.Label(label="Classification")
+        postprocess = external_utils.postprocess_label
+        examples = [["I feel great", "happy, sad", False]]
+        fn = external_utils.zero_shot_classification_wrapper(client)
+    # Example: sentence-transformers/distilbert-base-nli-stsb-mean-tokens
+    elif p == "sentence-similarity":
+        inputs = [
+            components.Textbox(
+                label="Source Sentence",
+                placeholder="Enter an original sentence",
             ),
-            "outputs": components.Dataframe(
-                label="Predictions", type="array", headers=["prediction"], render=False
+            components.Textbox(
+                lines=7,
+                placeholder="Sentences to compare to -- separate each sentence by a newline",
+                label="Sentences to compare to",
             ),
-            "preprocess": rows_to_cols,
-            "postprocess": lambda r: {
-                "headers": ["prediction"],
-                "data": [[pred] for pred in json.loads(r.text)],
-            },
-            "examples": example_data,
-        }
-
-    if p is None or p not in pipelines:
+        ]
+        outputs = components.JSON(label="Similarity scores")
+        examples = [["That is a happy person", "That person is very happy"]]
+        fn = external_utils.sentence_similarity_wrapper(client)
+    # Example: julien-c/ljspeech_tts_train_tacotron2_raw_phn_tacotron_g2p_en_no_space_train
+    elif p == "text-to-speech":
```

Comment:
Interesting, I'll check

Reply:
Ah so its because the InferenceClient returns a `bytes` object for this task. Its not trivial to infer the correct format from raw bytes, so I think its better to leave as a generic file. This is currently how we handle bytes in the `Audio.postprocess()` method -- we leave them as a generic file named "audio"